src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const parts = date.split(separator);
+
+  const dateParts = {};
+
+  fromFormat.forEach((part, index) => {
+    dateParts[part] = parts[index];
+  });
+
+  if (toFormat.includes('YY') && dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
+
+  if (toFormat.includes('YYYY') && dateParts['YY']) {
+    dateParts['YYYY'] =
+      dateParts['YY'] < '30' ? '20' + dateParts['YY'] : '19' + dateParts['YY'];
```

Comment:
The logic for determining the century for 'YY' might not be accurate for all cases. It assumes that any 'YY' less than '30' belongs to the 21st century, which might not be correct depending on the context. Ensure this logic aligns with the expected behavior.